connect/client.go

```diff
@@ -0,0 +1,108 @@
+package connect
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/url"
+	"time"
+
+	"google.golang.org/grpc"
+	grpc_backoff "google.golang.org/grpc/backoff"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/pomerium/zero-sdk/token"
+)
+
+type client struct {
+	endpoint    *url.URL
+	tokenCache  *token.Cache
+	minTokenTTL time.Duration
+	requireTLS  bool
+}
+
+func NewAuthorizedConnectClient(
+	ctx context.Context,
+	endpoint string,
+	cache *token.Cache,
+) (ConnectClient, error) {
+	url, err := url.Parse(endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing endpoint url: %w", err)
+	}
+
+	cc := &client{
+		tokenCache: cache,
+		endpoint:   url,
+		// streaming connection would reset based on token duration,
+		// so we need it be close to max duration 1hr
+		minTokenTTL: time.Minute * 55,
```

Comment:
I'm not sure I understand this; is there an advantage to setting it under 1 hour?

Reply:
OK, I think I understand now. I didn't understand the 1 hour maximum was from the authorization service.